### PR TITLE
Fix top navbar style

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I just realized that the display of the top navbar menu is broken (well, not good looking):

Screenshot from 2022-04-11 14-06-07

It comes from my changes in https://github.com/rust-lang/docs.rs/pull/1717 to allow having a different font size.

Overall, the style for the top navbar is quite the nightmare so I decided to remove the `float` properties and to use `flex` instead. The result is quite nice (imo) and fixes some small UI issues:

Screenshot from 2022-04-11 15-16-23

And it looks like this with bigger font size:

Screenshot from 2022-04-11 15-12-54

I checked other pages and they work well too.

### CODE REVIEW

**Comment by jsha** [comment]
In my local Chrome, with the light theme, it looks like this:

image

There are grey rectangles between the menu names. Looking the Dev Tools, it appears to be due to these rules in the light CSS:

```
::-webkit-scrollbar-thumb {
    background-color: rgba(36,37,39,0.6);
}
::-webkit-scrollbar-track {
    background-color: #ecebeb;
}
```

Those are getting applied to the `<div class="pure-menu pure-menu-horizontal">`, but I'm not sure why.

**Comment by GuillaumeGomez** [comment]
Great catch! I'll check tomorrow what's going on.

File: templates/style/_navbar.scss [hunk 1]
```diff
@@ -122,8 +128,21 @@ div.nav-container {
            See https://github.com/rust-lang/docs.rs/issues/1669.
            */
        .pure-menu-item a {
-            /* 0.5 em is the padding */
-            max-height: calc(#{$top-navbar-height} - 0.5em * 2);
+            max-height: 100%;
+            overflow-x: hidden;
```
Comment (jsha) [inline templates/style/_navbar.scss:132]: I think these lines are unnecessary. When I remove these properties locally, the unwanted grey rectangles (which are tiny vertical scrollbars) go away. Also, with these properties removed, even if the menu items' font size is weird, it doesn't interfere with the main page.

**Review by jsha — CHANGES_REQUESTED** [review]

File: templates/style/_navbar.scss (hunk 1 above)
Comment (jsha) [reply templates/style/_navbar.scss:132]: In fact, I think this whole CSS rule (including the comment about what happens with a locally installed old Fira Sans) should get deleted. Switching to the flex layout seems like it sufficed to fix the issue with overflowing menu titles, even without this rule.

Comment (jsha) [reply templates/style/_navbar.scss:132]: (I just tested in a locally installed Firefox, with the Fira Sans from https://github.com/rust-lang/docs.rs/issues/1669#issuecomment-1068396583, at these commits:

1. ad8d4fe33ba6d48669175221f3b651eb5e1e78ec (i.e. right before #1711)
2. 0ae58075652454f6bd535e39fdcd7be27f846fd5 (this branch)
3. 0ae58075652454f6bd535e39fdcd7be27f846fd5 with this CSS rule removed

And I can confirm that at (1) I have the overflow issue. Both (2) and (3) fix the overflow issue, and (3) additionally fixes the scrollbar issue introduced in this branch.

BTW the scrollbar issue seems to be Chrome-only.

File: templates/style/_navbar.scss [hunk 2]
```diff
@@ -28,6 +28,12 @@ div.nav-container {
     color: var(--color-navbar-standard);
     /* The font size must be specified in pixels because the height is specified in pixels. */
     font: 16px $font-family-sans;
+    position: relative;
+
+    .container, .pure-menu-horizontal {
+        position: relative;
+        height: 100%;
```
Comment (jsha) [inline templates/style/_navbar.scss:35]: What are the `position: relative` and `height: 100%` needed for here? I think they are unnecessary.

File: templates/style/_navbar.scss [hunk 3]
```diff
@@ -73,18 +79,18 @@ div.nav-container {
         }
     }
 
-    .pure-menu-right {
-        float: right;
-    }
-
     form.landing-search-form-nav {
         max-width: 1200px;
+        height: 100%;
```
Comment (jsha) [inline templates/style/_navbar.scss:84]: I think you can get rid of this `height: 100%`. That makes the search field vertically misaligned, but you can fix that by also removing the `height: 100%` on line 92.

File: templates/style/_navbar.scss [hunk 4]
```diff
@@ -73,18 +79,18 @@ div.nav-container {
...
+        display: flex;
+        flex-direction: row;
 
         #search-input-nav {
-            float: right;
             max-width: 150px;
             display: none;
             border-left: 1px solid var(--color-border);
+            height: 100%;
+            overflow-x: hidden;
```
Comment (jsha) [inline templates/style/_navbar.scss:93]: Why do we need `overflow-x: hidden` here?

File: templates/style/_navbar.scss [hunk 5]
```diff
@@ -122,8 +128,21 @@ div.nav-container {
...
+            text-overflow: ellipsis;
+       }
+
+       .docsrs-logo, .pure-menu-item a {
+            padding: 8px 1em;
+       }
+
+       .pure-menu-item {
+            height: 100%;
```
Comment (jsha) [inline templates/style/_navbar.scss:141]: I think this is also unnecessary.

**Review by jsha — CHANGES_REQUESTED** [review]

**Comment by jsha** [comment]
Overall, great change! I am really happy to have the items show up in the same HTML order as the visual order on the page.

File: templates/style/_navbar.scss (hunk 2 above)
Comment (GuillaumeGomez) [reply templates/style/_navbar.scss:35]: It's because with this, we can set the element of the navbar to `height: 100%`. It prevents the bottom margin to disappear in case the font size is too big.

File: templates/style/_navbar.scss (hunk 1 above)
Comment (GuillaumeGomez) [reply templates/style/_navbar.scss:132]: Without this rule:

Screenshot from 2022-04-13 16-51-54

With this rule:

Screenshot from 2022-04-13 16-52-21

So we definitely need it.

File: templates/style/_navbar.scss (hunk 3 above)
Comment (GuillaumeGomez) [reply templates/style/_navbar.scss:84]: Without the `height: 100%`:

Screenshot from 2022-04-13 16-56-07

**Comment by GuillaumeGomez** [comment]
I tested all of them and they all ended up in bad rendering in firefox. :-/

**Comment by jsha** [comment]
Which type of bad rendering? The "body text is too narrow" bad rendering? Or "bottom border of menu title is not visible"?

**Comment by GuillaumeGomez** [comment]
Size of top navbar items becomes chaotic:

Screenshot from 2022-04-13 16-56-07

For example, `sysinfo` item even gets its height different from the rest.

**Comment by jsha** [comment]
I spent some time fiddling with this locally and came up with a solution that I think is tidier, avoids the scrollbars on Chrome, and looks right on both Chrome and Firefox. I sent it as a PR against your branch: https://github.com/GuillaumeGomez/docs.rs/pull/1. Let me know what you think!

**Comment by GuillaumeGomez** [comment]
Nice, thanks a lot! Will test it locally as soon as possible.

**Comment by GuillaumeGomez** [comment]
I integrated @jsha's improvements so this should now be ready!

**Comment by jsha** [comment]
I've manually tested the version on this branch in Firefox and Chrome, with the old version of Fira Sans locally installed. Tests I did:

 - Scrolled vertically to make sure the navbar stayed in place
 - Opened and closed menus
 - Verified that the rustdoc content was not artificially narrow
 - Verified that the vertical bars separating menu titles went all the way to the top and bottom of the navbar
 - Verified that menu titles were approximately centered vertically

I also tested with a large font size in firefox, and a large minimum font size. The presentation wasn't perfect but it was good enough. And it didn't interere with the rustdoc content.

**Review by syphar — APPROVED** [review]
LGTM, definitely fixes #1725
